src/datasources/balances-api/balances-api.manager.ts

```diff
@@ -0,0 +1,34 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { IValkBalancesApi } from '@/datasources/balances-api/valk-balances-api.service';
+import { IBalancesApi } from '@/domain/interfaces/balances-api.interface';
+import { IBalancesApiManager } from '@/domain/interfaces/balances-api.manager.interface';
+import { Inject, Injectable } from '@nestjs/common';
+
+@Injectable()
+export class BalancesApiManager implements IBalancesApiManager {
+  private readonly valkBalancesChainIds: string[];
+  private readonly valkBalancesApi: IBalancesApi;
+
+  constructor(
+    @Inject(IConfigurationService)
+    private readonly configurationService: IConfigurationService,
+    @Inject(IValkBalancesApi) valkBalancesApi: IBalancesApi,
+  ) {
+    this.valkBalancesChainIds = this.configurationService.getOrThrow<string[]>(
+      'features.valkBalancesChainIds',
+    );
+    this.valkBalancesApi = valkBalancesApi;
+  }
+
+  isExternalized(chainId: string): boolean {
+    const allProvidersChainIds = this.valkBalancesChainIds;
+    return allProvidersChainIds.includes(chainId);
+  }
+
+  getBalancesApi(chainId: string): IBalancesApi {
+    if (this.valkBalancesChainIds.includes(chainId)) {
```

Comment:
My intention here was to implement something like a _router_ so the execution would go through each provider and check if the `chainId` is included in the chain IDs the provider manages. At this point, as only one provider is available I agree we could reuse the other function in the interface. Still, it may be clearer to check the provider IDs explicitly to understand the purpose of the function (and also keeping in mind that the cost is the same), wdyt?

Reply:
Makes sense to me - you need not include my suggestion. We should instead add a comment or include provider-specific helpers, e.g. `isSupportByValk` to be more declarative.